From 0e694a34ff5637de02a5ff61447748164536d75e Mon Sep 17 00:00:00 2001
From: Ramesh Babu <ramesh.babu@intel.com>
Date: Mon, 6 Jun 2016 13:13:15 +0530
Subject: [PATCH 1994/2508] ALSA: hda: ext: add spib to stream context

commit e523162ccd7b92bd8d1f99bd8d02ef2b2539992c from
https://github.com/01org/linux-apollolake-i

Need to save spib value in stream context
during stream suspend/resume operations.

Change-Id: I268d235b72a27aeae4284b5c855dd6ac8d96b411
Signed-off-by: Ramesh Babu <ramesh.babu@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/9565
---
 include/sound/hdaudio_ext.h     |    1 +
 sound/hda/ext/hdac_ext_stream.c |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/sound/hdaudio_ext.h b/include/sound/hdaudio_ext.h
index a83a19b..72310ae 100644
--- a/include/sound/hdaudio_ext.h
+++ b/include/sound/hdaudio_ext.h
@@ -93,6 +93,7 @@ struct hdac_ext_stream {
 
 	u32 dpib;
 	u32 lpib;
+	u32 spib;
 	bool decoupled:1;
 	bool link_locked:1;
 	bool link_prepared;
diff --git a/sound/hda/ext/hdac_ext_stream.c b/sound/hda/ext/hdac_ext_stream.c
index 26f4b8f..119611ee 100644
--- a/sound/hda/ext/hdac_ext_stream.c
+++ b/sound/hda/ext/hdac_ext_stream.c
@@ -460,6 +460,8 @@ int snd_hdac_ext_stream_set_spib(struct hdac_ext_bus *ebus,
 	}
 
 	writel(value, stream->spib_addr);
+	/* save the value in stream context */
+	stream->spib = value;
 
 	return 0;
 }
-- 
1.7.5.4

